mlir/test/Target/LLVMIR/Import/metadata-tbaa-struct.ll

```diff
@@ -0,0 +1,21 @@
+; RUN: mlir-translate -import-llvm %s | FileCheck %s
+
+!0 = !{!1, !1, i64 0}
+!1 = !{!"scalar type", !2, i64 0}
+!2 = !{}
+
+!3 = !{i64 0, i64 4, !0, i64 8, i64 4, !0}
+
+; CHECK: llvm.func @memcpy_with_tbaa_struct(
+; CHECK-SAME: %[[ARG:.*]]: !llvm.ptr) {
+; CHECK:     %[[C4:.*]] = llvm.mlir.constant(4 : i32) : i32
+; CHECK: "llvm.intr.memcpy"(%[[ARG]], %[[ARG]], %[[C4]]) <{isVolatile = false,
+; CHECK-SAME: tbaa_struct = #llvm.tbaa_struct_tag<members = {<0, 4, #tbaa_tag>, <8, 4, #tbaa_tag>}>}>
+; CHECK-SAME: (!llvm.ptr, !llvm.ptr, i32) -> ()
```

Comment:
Good catch.
IIRC, even if there was multiple uses of the struct it would be inlined. So we would need to specify it to not be inlined. Can add that when I get a chance

Reply:
Yes if it is not too much work, feel free to change it. But it is minor. I am more curious what triggers that behaviour.

Reply:
I remember a similar question was asked a while ago, so I thought I could do a quick google/discord search for it. 

It appears that it is controlled with [`OpAsmDialectInterface`](https://mlir.llvm.org/doxygen/classmlir_1_1OpAsmDialectInterface.html), but it is not quite clear as to why it does not use the alias from `getAlias`.

Alternatively it seems you could also implement the interface method for the `memcpy` operation such that it would always alias the struct. 

But looking further into those wouldn't be trivial, so I will leave it for now.

Reply:
Ok thanks for looking at that, no worries you can keep it as it is.